src/openforms/submissions/tasks/__init__.py

```diff
@@ -22,127 +21,50 @@
 logger = logging.getLogger(__name__)
 
 
-def on_completion(submission_id: int) -> None:
-    """
-    Celery chain of tasks to execute on a submission completion event.
+def on_post_submission_event(submission_id: int) -> None:
+    # this can run any time because they have been claimed earlier
+    cleanup_temporary_files_for.delay(submission_id)
 
-    This SHOULD be invoked as a transaction.on_commit(...) handler, therefore it should
-    not execute any extra queries in the process this function is running in.
-    """
-    # use immutable signatures so that the result of previous tasks is not passed
-    # in as an argument to chained tasks
+    # If the form involves appointments and no appointment has been scheduled yet, schedule it.
+    # Todo: deprecated => Not needed with the new appointment flow
     register_appointment_task = maybe_register_appointment.si(submission_id)
+
+    # Perform any pre-registration task specified by the registration plugin. If no registration plugin is configured,
+    # just set a submission reference (if it hasn't already been set)
     pre_registration_task = pre_registration.si(submission_id)
-    send_email_cosigner_task = send_email_cosigner.si(submission_id)
+
+    # Generate the submission report. Contains information about the payment and co-sign status.
     generate_report_task = generate_submission_report.si(submission_id)
+
+    # Attempt registering submission
     register_submission_task = register_submission.si(submission_id)
-    finalize_completion_task = finalize_completion.si(submission_id)
-
-    # for the orchestration with distributed processing and dependencies between
-    # tasks, see the Celery documentation:
-    # https://docs.celeryproject.org/en/stable/userguide/canvas.html#guide-canvas
-    # Additionally, check the developer documentation of Open Forms for a visual
-    # representation of the task orchestration and dependencies (TODO, @joeri)
-    # The linked task (= next task) is only executed if the previous task returns
-    # successfully, so error handling needs to happen inside each task.
-    on_completion_chain = chain(
-        # The appointment must be registered before a confirmation PDF is generated and
-        # any backend registration happens, as on-failure, the user should get feedback
-        # about the failure.
+
+    # Payment status update
+    # Todo: deprecated => Not needed with the new payment flow
```

Comment:
I was under the impression that both flows are equally viable - payment prevents registration or payment status update afterwards, so should we actually deprecate this? If there's a "mode select" in the global configuration or at the form level, deprecating it might be a bit too eager.

Reply:
Ahh I had understood that we eventually wanted to remove it. I will remove the comment.